src/main/docs/guide/cli/usage.adoc

```diff
@@ -0,0 +1,76 @@
+== Creating keypairs
+
+A utility to help with creating keypairs. This is akin to doing something like so with openssl
+
+[source,bash]
+----
+$ openssl genrsa -out /tmp/mydomain.com-key.pem 4096
+----
+
+These keypairs will be used for both ACME accounts as well as each domain will also need its own keypair defined.
+
+Usage:
+
+[source,bash]
+----
+Usage: acme-cli create-key [-h] [-k=<keyDir>] [-n=<keyName>] [-s=<keySize>]
+Creates an keypair for use with account creation
+  -h, --help                 Show usage of this command
+  -k, --key-dir=<keyDir>     Custom location on disk to put the key to be used
+                               with this account.
+                               Default: /tmp
+  -n, --key-name=<keyName>   Name of the key to be created
+                               Default: acme.pem
+  -s, --key-size=<keySize>   Size of the key to be generated
+                               Default: 4096
+----
+
+== Creating an Account
+
+Creates a new account for a given ACME provider. This command will either create a new account keypair for you or you can pass
+the account keypair that you have generated using the `acme-cli create-key` or via `openssl` or other means in as a parameter.
+
+https://certbot.eff.org/[Certbot] or many of the other tools out there can also accomplish this step if you dont want to use this tool.
+
+Usage:
+
+[source,bash]
+----
+Usage: acme-cli create-account [-h] -e=<email> [-k=<keyDir>] [-n=<keyName>] -u=<serverUrl>
+Creates a new account on the given ACME server
+  -e, --email=<email>        Email address to create account with.
+  -h, --help                 Show usage of this command
+  -k, --key-dir=<keyDir>     Directory to create/find the key to be used for this account.
+                               Default: /tmp
+  -n, --key-name=<keyName>   Name of the key to be created/used
+                               Default: acme.pem
+  -u, --url=<serverUrl>      Location of acme server to use.
+                             Let's Encrypt Prod :
+                             https://acme-v02.api.letsencrypt.org/directory
+                             Let's Encrypt Staging :
+                             https://acme-staging-v02.api.letsencrypt.org/directory
+                               Default: null
+----
+
+== Deactivating an Account
+
+Deactivates a given account based on the account key that was used to create the account.
+
+Usage:
+
+[source,bash]
+----
+Usage: acme-cli deactivate-account [-h] [-k=<keyDir>] [-n=<keyName>] -u=<serverUrl>
+Deactivates an existing ACME account
+  -h, --help                 Show usage of this command
+  -k, --key-dir=<keyDir>     Directory to find the key to be used for this account.
+                               Default: /tmp
+  -n, --key-name=<keyName>   Name of the key to be used
+                               Default: acme.pem
+  -u, --url=<serverUrl>      Location of acme server to use.
```

Comment:
i wonder how difficult it would be to add exclusive `--lets-encrypt-prod`/`--lets-encrypt-staging

Reply:
Turns out it would be roughly this difficult: https://github.com/micronaut-projects/micronaut-acme/commit/cfd6d43c60860aacfd629e69e1328a58b51b2576

Reply:
![](https://ignitedmoth.files.wordpress.com/2017/02/d871ppy.gif?w=840) .... i'll cherry pick that over and finish up the documentation 👍